enterprise/backend/src/metabase_enterprise/semantic_search/embedding.clj

```diff
@@ -76,26 +149,42 @@
 
 (defrecord OpenAIProvider []
   EmbeddingProvider
-  (-get-embedding [_ text]
+  (-get-embedding [this text]
     (try
       (log/info "Generating OpenAI embedding for text of length:" (count text))
-      (let [api-key (semantic-settings/openai-api-key)
-            model (get-model)
-            endpoint "https://api.openai.com/v1/embeddings"]
-        (when-not api-key
-          (throw (ex-info "OpenAI API key not configured" {:setting "ee-openai-api-key"})))
-        (-> (http/post endpoint
-                       {:headers {"Content-Type" "application/json"
-                                  "Authorization" (str "Bearer " api-key)}
-                        :body    (json/encode {:model model
-                                               :input text})})
-            :body
-            (json/decode true)
-            (get-in [:data 0 :embedding])))
+      (first (-get-embeddings-batch this [text]))
       (catch Exception e
         (log/error e "Failed to generate OpenAI embedding for text of length:" (count text))
         (throw e))))
 
+  (-get-embeddings-batch [_ texts]
+    (let [batches (create-batches texts)
+          batch-results (mapv (fn [batch-texts]
+                                (try
+                                  (log/info "Generating" (count batch-texts) "OpenAI embeddings in batch")
```

Comment:
maybe `debug` or `trace` level? I think I did `trace` for index batch updates, but maybe that's too low. In any case I can look at cleaning this stuff up / unifying log levels when I look at monitoring.